src/lambda/wizards/samInitWizard.ts

```diff
@@ -80,6 +100,171 @@ export class DefaultCreateNewSamAppWizardContext extends WizardContext implement
         return val ? (val.label as Runtime) : undefined
     }
 
+    public async promptUserForTemplate(
+        currRuntime: Runtime,
+        currTemplate?: SamTemplate
+    ): Promise<SamTemplate | undefined> {
+        const templates = getSamTemplateWizardOption(currRuntime)
+        const quickPick = picker.createQuickPick<vscode.QuickPickItem>({
+            options: {
+                ignoreFocusOut: true,
+                title: localize('AWS.samcli.initWizard.template.prompt', 'Select a SAM Application Template'),
+                value: currTemplate ? currTemplate : ''
+            },
+            buttons: [this.helpButton, vscode.QuickInputButtons.Back],
+            items: templates.toArray().map(template => ({
+                label: template,
+                alwaysShow: template === currTemplate,
+                description:
+                    template === currTemplate
+                        ? localize('AWS.wizard.selectedPreviously', 'Selected Previously')
+                        : getTemplateDescription(template)
+            }))
+        })
+
+        const choices = await picker.promptUser({
+            picker: quickPick,
+            onDidTriggerButton: (button, resolve, reject) => {
+                if (button === vscode.QuickInputButtons.Back) {
+                    resolve(undefined)
+                } else if (button === this.helpButton) {
+                    vscode.env.openExternal(vscode.Uri.parse(samInitDocUrl))
+                }
+            }
+        })
+        const val = picker.verifySinglePickerOutput(choices)
+
+        //eventBridgeStarterAppTemplate requires aws credentials
+        if (val && val.label === eventBridgeStarterAppTemplate) {
+            if (this.credentialsNotFound) {
+                await ext.awsContextCommands.onCommandLogin()
+
+                return exitWizard
+            }
+
+            //check if detected samCliVersion supports eventBridgeStarterAppTemplate
+            if (semver.lt(this.samCliVersion, MINIMUM_SAM_CLI_VERSION_SCHEMAS_SUPPORT_INCLUSIVE)) {
```

Comment:
It would be nice to (append) decorate the QuickPick Item with a status saying something like "Requires newer SAM CLI". Then if they get this message, there's less surprise.

The notification could also contain a "Get SAM CLI" button which takes the user to the sam cli website. See actionGoToSamCli in "src\shared\sam\cli\samCliValidationNotification.ts" for sample code (label text and url launch).

Reply:
Got approval just to upgrade minimum required sam cli version in our retro meeting, will bump it to 0.38.0. This will get rid of this check, resolving.